release/pkg/controller/operator/controller.go

```diff
@@ -0,0 +1,250 @@
+		docker:   registry.MustDockerRunner(),
+		validate: true,
+		publish:  true,
+	}
+	for _, opt := range opts {
+		if err := opt(o); err != nil {
+			logrus.WithError(err).Fatal("Failed to apply option")
+		}
+	}
+
+	if o.repoRoot == "" {
+		logrus.Fatal("No repository root specified")
+	}
+
+	return o
+}
+
+func (o *OperatorController) Build(outputDir string) error {
+	if !o.isHashRelease {
+		return fmt.Errorf("operator controller builds only for hash releases")
+	}
+	if o.validate {
+		if err := o.PreBuildValidation(outputDir); err != nil {
+			return err
+		}
+	}
+	component, componentsVersionPath, err := hashrelease.GenerateOperatorComponents(outputDir)
+	if err != nil {
+		return err
+	}
+	env := os.Environ()
+	env = append(env, fmt.Sprintf("OS_VERSIONS=%s", componentsVersionPath))
+	env = append(env, fmt.Sprintf("COMMON_VERSIONS=%s", componentsVersionPath))
+	if _, err := o.make("gen-versions", env); err != nil {
+		return err
+	}
+	env = os.Environ()
+	env = append(env, fmt.Sprintf("ARCHES=%s", strings.Join(o.architectures, " ")))
+	env = append(env, fmt.Sprintf("VERSION=%s", component.Version))
+	env = append(env, fmt.Sprintf("BUILD_IMAGE=%s", component.Image))
+	if _, err := o.make("image-all", env); err != nil {
+		return err
+	}
+	for _, arch := range o.architectures {
+		currentTag := fmt.Sprintf("%s:latest-%s", component.Image, arch)
+		newTag := fmt.Sprintf("%s-%s", component.String(), arch)
+		if err := o.docker.TagImage(currentTag, newTag); err != nil {
+			return err
+		}
+	}
+	env = os.Environ()
+	env = append(env, fmt.Sprintf("VERSION=%s", component.Version))
+	env = append(env, fmt.Sprintf("BUILD_IMAGE=%s", component.Image))
+	env = append(env, fmt.Sprintf("BUILD_INIT_IMAGE=%s", component.InitImage().Image))
+	if _, err := o.make("image-init", env); err != nil {
+		return err
+	}
+	currentTag := fmt.Sprintf("%s:latest", component.InitImage().Image)
+	newTag := component.InitImage().String()
+	return o.docker.TagImage(currentTag, newTag)
+}
+
+func (o *OperatorController) PreBuildValidation(outputDir string) error {
+	if !o.isHashRelease {
+		return fmt.Errorf("operator controller builds only for hash releases")
+	}
+	if len(o.architectures) == 0 {
+		return fmt.Errorf("no architectures specified")
+	}
+	operatorComponent, err := hashrelease.RetrievePinnedOperator(outputDir)
+	if err != nil {
+		return err
+	}
+	if operatorComponent.Version != o.version {
+		return fmt.Errorf("operator version does not match the pinned version")
+	}
+	return nil
+}
+
+func (o *OperatorController) Publish(outputDir string) error {
+	if o.validate {
+		if err := o.PrePublishValidation(); err != nil {
+			return err
+		}
+	}
+	operatorComponent, err := hashrelease.RetrievePinnedOperator(outputDir)
+	if err != nil {
+		logrus.WithError(err).Error("Failed to get operator component")
+		return err
+	}
+	var imageList []string
+	for _, arch := range o.architectures {
+		imgName := fmt.Sprintf("%s-%s", operatorComponent.String(), arch)
```

Comment:
```suggestion
		imgName := fmt.Sprintf("%s-%s", operatorComponent.String(), arch)
		logCtx := logrus.WithField("image", imgName)
```

Reply:
You can initialize a context logger once and then use it below rather than repeating the `WithField` on every log.